bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/NeeoRoomProtocol.java

```diff
@@ -263,19 +263,14 @@ private void refreshActiveScenarios() {
             logger.debug("API is null [likely bridge is offline]");
         } else {
             try {
-                final String[] activeScenarios = api.getActiveScenarios();
-                final String[] oldScenarios = this.activeScenarios.getAndSet(activeScenarios);
-
-                if (!ArrayUtils.isEquals(activeScenarios, oldScenarios)) {
-                    for (String scenario : activeScenarios) {
-                        refreshScenarioStatus(scenario);
-                    }
-
-                    for (String oldScenario : oldScenarios) {
-                        if (!ArrayUtils.contains(activeScenarios, oldScenario)) {
-                            refreshScenarioStatus(oldScenario);
-                        }
-                    }
+                final String[] apiScenarios = api.getActiveScenarios();
+                final List<String> activeScenarios = Arrays.asList(apiScenarios);
+                final List<String> oldScenarios = Arrays.asList(this.activeScenarios.getAndSet(apiScenarios));
+
+                if (!activeScenarios.equals(oldScenarios)) {
+                    activeScenarios.forEach(this::refreshScenarioStatus);
+                    oldScenarios.removeIf(activeScenarios::contains);
```

Comment:
Aren't the lists returned by `Arrays.asList` immutable? I don't think this would work unless you make a copy of the `oldScenarios ` List first.